### PR TITLE
Adding timestamp functions into potential problems for qual tool

### LINKED ISSUE #6645: [FEA] Qualification Tool: Print timestamp related functions.
I wish Qualification Tool can print timestamp related functions such as current_timestamp.
This can help us know in advance what timestamp related function are used by this job, just in case if user are running non-UTC timezone.

### PR DESCRIPTION
**Pull request by mattahrens**
Signed-off-by: Matt Ahrens <matthewahrens@gmail.com>

Closes #6645 

Adding timestamp functions in potential problems of qual tool output, with TIMEZONE keyword along with function name.

Items for review:
- confirming list of timestamp functions to include
- confirming if output in potential problems section is as desired

Current implementation would add text in the Potential Problems column of CSV output with info that lists each function related to timezone separately: `TIMEZONE to_timestamp():TIMEZONE hour():TIMEZONE current_timestamp():TIMEZONE second()`

### CODE REVIEW

**Comment by mattahrens** [comment]
build

**Comment by mattahrens** [comment]
build

**Comment by mattahrens** [comment]
build

File: tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/QualificationSuite.scala [hunk 1]
```diff
@@ -1045,6 +1045,44 @@ class QualificationSuite extends FunSuite with BeforeAndAfterEach with Logging {
       }
     }
   }
+
+  test("test potential problems timestamp") {
+    TrampolineUtil.withTempDir { eventLogDir =>
+      val listener = new ToolTestListener
```
Comment (tgravescs) [inline tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/QualificationSuite.scala:1051]: don't need the listener for this test

File: tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala [hunk 2]
```diff
@@ -208,14 +208,22 @@ abstract class AppBase(
   }
 
   private val UDFRegex = ".*UDF.*"
-  private val UDFKeywords = Map(UDFRegex -> "UDF")
+
+  private val potentialIssuesRegexMap = Map(
+    UDFRegex -> "UDF",
```
Comment (tgravescs) [inline tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala:213]: I'm wondering if TimeAdd shows up in the plans or if it just shows as a +, I think its just a + with a cast so might not be easy to catch here.

Comment (tgravescs) [reply tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala:213]: https://github.com/NVIDIA/spark-rapids/commit/7910229d025fab20081f54b28618f3db2be35225

After testing a few things, it doesn't shows up in the plan, I think it just shows up as +,  so would be hard to find.

File: tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/QualificationSuite.scala (hunk 1 above)
Comment (mattahrens) [reply tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/QualificationSuite.scala]: resolved in commit `f76108d`

File: tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/QualificationSuite.scala [hunk 3]
```diff
@@ -1045,6 +1045,44 @@ class QualificationSuite extends FunSuite with BeforeAndAfterEach with Logging {
...
+        // so create a new one to read in the csv file
+        createSparkSession()
+
+        // validate that the SQL description in the csv file escapes commas properly
+        val outputResults = s"$outpath/rapids_4_spark_qualification_output/" +
+          s"rapids_4_spark_qualification_output.csv"
+        val outputActual = readExpectedFile(new File(outputResults))
+        assert(outputActual.select("Potential Problems").first.getString(0) == 
+          "TIMEZONE to_timestamp():TIMEZONE hour():TIMEZONE current_timestamp():TIMEZONE second()")
+        assert(outputActual.collect().size == 1)
```
[the remark this answers is not in the document]
Comment (mattahrens) [reply tools/src/test/scala/com/nvidia/spark/rapids/tool/qualification/QualificationSuite.scala]: good feedback.  resolved in commit `f76108d`

**Review by nartal1 — APPROVED** [review]

**Comment by mattahrens** [comment]
build

**Pull request merged**